baseapp/server.go

```diff
@@ -130,6 +135,35 @@ func (s *Server) Start() error {
 	return s.server.ListenAndServe()
 }
 
+// StartAndStopGracefully starts the server, blocks, and shutdowns gracefully
+func (s *Server) StartAndStopGracefully() {
```

Comment:
I'm not 100% on the usability of having both Start and StartAndStopGracefully. But I also don't want to introduce an API breaking change here, and I think its still useful to keep Start if any clients want to implement their own graceful shutdown logic.

Reply:
I think there is a valid use case for both. In many cases we just don't care how the server shuts down and a default implementation, in the form of `StartAndStopGracefully`, is good. However, there might be cases when you want to do something more custom. `Start` allows for this behaviour, and I think is a better than having some form of overridable function

Reply:
Allowing people to opt-in or opt-out makes sense, but I don't really like the current method split. I also don't think it's necessarily bad to change the behavior of `Start()` as long as the signature doesn't change and the new behavior is widely useful. Some alternate ideas:

1. Only have `Start()` but have it behave differently based on the value of `ShutdownWaitTime`. If the value is `nil`, keep the old behavior, but if it is non-nil, then enable graceful shutdown (possibly with a wait time of zero)

2. Have `Start()` implement graceful shutdown by calling new methods `StartAsync()` and `WaitForShutdown()`. Clients that can't support graceful shutdown (which isn't any of our apps) can switch to calling `StartAsync()` and then implementing their own shutdown or blocking logic. Or do the reverse: keep `Start()` as is, but add the two new methods as the way to opt-in to graceful shutdown.

The main idea for me is that we either have a single monolithic method that implements behavior internally, or we add multiple methods that let users choose behavior by calling them in sequence.

And ultimately, I think graceful shutdown is the more correct behavior, so I'd lean towards having that as the default.

Reply:
I decided to go with 1, let me know what you think.